tests/test_app.py

```diff
@@ -51,6 +51,27 @@ def test_metrics(client, now, enrollment):
         match_querystring=True,
         json=sample_data
     )
+    metric_name = b'my_costs'
+    app.config['METRIC_NAME'] = metric_name
+
+    rsp = client.get('/metrics')
+    assert rsp.status_code == 200
+    assert rsp.data.count(metric_name) == 4
+
+
+@responses.activate
+def test_metrics_name_default_value(client, now, enrollment):
+    responses.add(
+        method='GET',
+        url="https://ea.azure.com/rest/{0}/usage-report?month={1}&type=detail&fmt=Json".format(enrollment, now),
+        match_querystring=True,
+        json=sample_data
+    )
+    #make sure that app.config is empty
```

Comment:
I think I start to understand: The `app` object is a global object, isn't it? So any modification in any test will potentially influence other tests?

In this case, I would strongly advice to shift the responsibility to establish a well-defined state of `app.config` to the fixture. That would include a reset of sorts before adding the desired configuration options. Failing to do this basically couples your tests with each other, and test results may depend on the order of test execution. The latter could be caused by the switch from Python 2 to 3.